pennylane_sf/hw.py

```
@@ -0,0 +1,134 @@
+# limitations under the License.
+"""
+Strawberry Fields Hardware device
+=================================
+
+**Module name:** :mod:`pennylane_sf.hw`
+
+.. currentmodule:: pennylane_sf.hw
+
+The Strawberry Fields Hardware plugin implements all the :class:`~pennylane.device.Device` methods,
+and provides access to Xanadu's continuous-variable quantum hardware.
+
+Classes
+-------
+
+.. autosummary::
+   StrawberryFieldsRemote
+
+
+Code details
+~~~~~~~~~~~~
+"""
+
+from collections import OrderedDict
+
+import numpy as np
+
+import strawberryfields as sf
+
+# import gates
+from strawberryfields.ops import BSgate, Interferometer, MeasureFock, Rgate, S2gate
+from strawberryfields.utils.post_processing import (
+    all_fock_probs_pnr,
+    samples_expectation,
+    samples_variance,
+)
+
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsRemote(StrawberryFieldsSimulator):
+    r"""StrawberryFields remote device for PennyLane.
+
+    A valid Strawberry Fields API token is needed for access. This token can be
+    passed when creating the device. The default configuration options of
+    Strawberry Fields are used to store the authentication token in a
+    configuration file.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+        shots (int): number of circuit evaluations/random samples used to
+            estimate expectation values of observables
+        chip (str): name of the remote chip to be used
+        hbar (float): the convention chosen in the canonical commutation
+            relation :math:`[x, p] = i \hbar`
+        sf_token (str): the SF API token used for remote access
+    """
+    name = "Strawberry Fields Hardware PennyLane plugin"
+    short_name = "strawberryfields.ai"
+
+    _operation_map = {
+        "Beamsplitter": BSgate,
+        "Rotation": Rgate,
+        "TwoModeSqueezing": S2gate,
+        "Interferometer": Interferometer,
+    }
```

@antalszava, I would be tempted to let this device accept the same operations as the fock/gaussian simulators. This way, if a new backend comes online that supports additional operations, we don't need to update this class.

If a user tries to use a operation that is not supported by the backend, the SF compiler should raise an error.